Update FileSaver.js

vnatara opened the pull request:
Remove auto_bom function. This will stop the BOM encoding issue.

vnatara committed:
Update FileSaver.js

Commented out 2 places where the auto_bom function is called. This will stop the BOM encoding issue.

markfullmer reviewed (changes requested):
Hi @vnatara ! Thanks for submitting this pull request! I'll be testing this momentarily on MacOS and Windows just to verify that the removal of the BOM character isn't problematic for those operating systems (I don't see why it would be, but always good to test!)

As far as the actual change, I note that both instances (lines 78 & 152) are wrapped in a conditional check for `if (!no_auto_bom) {`.

If there are no longer any instances where we do want the BOM to be prepended (see the [function that performs this here](https://github.com/writecrow/ciabatta/blob/c37bfe7cba7e52774bbc50d8af60bc4e00116acb/manual_deidentification/interactive_de-identifying_tool/js/FileSaver.js#L68-L74), then I think we can:

- [x] Remove the conditionals completely ([here](https://github.com/writecrow/ciabatta/blob/master/manual_deidentification/interactive_de-identifying_tool/js/FileSaver.js#L77-L79) and [here](https://github.com/writecrow/ciabatta/blob/master/manual_deidentification/interactive_de-identifying_tool/js/FileSaver.js#L151-L153))
- [x] Remove the `no_auto_bom` parameter from the two functions that reference it ([here](https://github.com/writecrow/ciabatta/blob/master/manual_deidentification/interactive_de-identifying_tool/js/FileSaver.js#L76) and [here](https://github.com/writecrow/ciabatta/blob/master/manual_deidentification/interactive_de-identifying_tool/js/FileSaver.js#L142))
- [x] Remove the helper function that prepends the BOM character ([here](https://github.com/writecrow/ciabatta/blob/master/manual_deidentification/interactive_de-identifying_tool/js/FileSaver.js#L68-L75))

Additionally, since we're using Git to keep track of changes to this tool, the commit message will be our point of reference of what changed & why. The first commit message, `Commented out 2 places where the auto_bom function is called. This will stop the BOM encoding issue.` is pretty sufficient at explaining what is changing and why, so if you agree that we can make the modifications I suggested above, then let's capture that in the commit message (e.g., something along the lines of "Remove methodology for prepending BOM character, since exports should always be in UTF-8 encoding."

vnatara committed:
Update FileSaver.js

Entirely remove methodology for prepending BOM character, since exports should always be in UTF-8 encoding.

vnatara commented:
Everything you said makes sense to me, so I've added another commit with the changes you suggested, like removing the auto_bom function entirely. From my testing, it seems to work fine but you may want to double check before you merge it. Thanks!

markfullmer reviewed (approved):
The code changes look good, and this functionally checks out, @vnatara 

1. Using a suggestion at https://blog.ijun.org/2011/01/to-check-if-file-contains-utf-8-bom-at.html , I exported a file using the current code and your changes, and the output corroborated that the original approach would include the BOM character, while the change avoids it:

![Screenshot from 2022-04-23 12-53-59](https://user-images.githubusercontent.com/4305692/164934064-7a2677d6-70f8-445a-b372-a2da8e22662c.png)

I also tested the various options in the tool and found no regressions due to this change. 

Thanks for the contribution!
